src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -343,9 +413,25 @@ void setNotifier(Notifier* n) {
 */
 DBClientBase* getMongoConnection(void)
 {
-    return connection;
+#ifdef UNIT_TEST
+  return connection;
+#else
+  return mongoPoolConnectionGet();
+#endif
 }
 
+DBClientBase* mongoInitialConnectionGetForUnitTest(void)
+{
+  return mongoPoolConnectionGet();
+}
+
+void releaseMongoConnection(DBClientBase* connection)
```

Comment:
Ok, let's keep the function. What about making explicit that for unit test no release operation is needed?

```
void releaseMongoConnection(DBClientBase* connection)
#ifndef UNIT_TEST
 mongoPoolConnectionRelease(connection)
#endif
 }
```